code/game/turfs/closed/wall/reinf_walls.dm

```diff
@@ -230,9 +230,9 @@
 /turf/closed/wall/r_wall/rust_heretic_act()
 	if(prob(50))
 		return
-	if(prob(70))
-		new /obj/effect/temp_visual/glowing_rune(src)
-	ChangeTurf(/turf/closed/wall/r_wall/rust)
+	if(GetComponent(/datum/component/rust))
```

Comment:
Don't use `GetComponent` like this, it's generally frowned upon--instead have the rust component just apply a trait that is rusted.